diff -rupN mod-cband-0.9.6.1/Makefile.in mod-cband-0.9.6.1-patched/Makefile.in
--- mod-cband-0.9.6.1/Makefile.in       2005-12-23 14:57:00.000000000 +0100
+++ mod-cband-0.9.6.1-patched/Makefile.in       2013-11-08 11:04:41.257331214 +0100
@@ -21,7 +21,7 @@
 #

 APXS=@APXS@
-APXS_OPTS=-Wc,-Wall -Wc,-DDST_CLASS=@DST_CLASS@
+APXS_OPTS=-lm -Wc,-Wall -Wc,-DDST_CLASS=@DST_CLASS@
 SRC=src/mod_cband.c
 OBJ=src/.libs/mod_cband.so

diff -rupN mod-cband-0.9.6.1/src/mod_cband.c mod-cband-0.9.6.1-patched/src/mod_cband.c
--- mod-cband-0.9.6.1/src/mod_cband.c   2006-01-22 02:23:27.000000000 +0100
+++ mod-cband-0.9.6.1-patched/src/mod_cband.c   2013-11-08 11:07:53.813324452 +0100
@@ -1278,7 +1278,7 @@ int mod_cband_get_dst(request_rec *r)
     p.bitlen = 32;
     p.ref_count = 0;
     p.family = AF_INET;
-    p.add.sin.s_addr = inet_addr(r->connection->remote_ip);
+    p.add.sin.s_addr = inet_addr(r->connection->client_ip);

     node = patricia_search_best(config->tree, &p);

@@ -1287,7 +1287,7 @@ int mod_cband_get_dst(request_rec *r)

         if (leaf) {
 #ifdef DEBUG
-            fprintf(stderr,"%s leaf %s\n",r->connection->remote_ip,leaf);
+            fprintf(stderr,"%s leaf %s\n",r->connection->client_ip,leaf);
             fflush(stderr);
 #endif
            return atoi(leaf);
@@ -1307,10 +1307,10 @@ int mod_cband_get_remote_host(struct con
     if (entry == NULL)
        return -1;

-    if (c->remote_ip != NULL)
-       addr = inet_addr(c->remote_ip);    
+    if (c->client_ip != NULL)
+       addr = inet_addr(c->client_ip);    
     else
-       addr = c->remote_addr->sa.sin.sin_addr.s_addr;
+       addr = c->client_addr->sa.sin.sin_addr.s_addr;

     time_now = apr_time_now();
     hosts = config->remote_hosts.hosts;
